src/parse.rs

```diff
@@ -102,31 +95,59 @@ pub enum ParseResult {
     Unsupported(SentenceType),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum NmeaError<'a> {
+    /// The provided input was not a proper UTF-8 string
+    Utf8DecodingError,
+    /// The checksum of the sentence was corrupt or wrong
+    ChecksumMismatch,
```

Comment:
It would be nice to have expected and calculated checksum. Some times it helps understand problem if you have original NMEA message.
For example it possible to see if one bit changed, then it is the most likely memory bit flip error,
if there are more bits changed then it is the most likely transport layer problem.

Reply:
Sounds very much reasonable, I'll add that!